Sources/Datadog/RUM/AutoInstrumentation/Views/UIKitRUMViewsPredicate.swift

```diff
@@ -58,15 +58,20 @@ public struct DefaultUIKitRUMViewsPredicate: UIKitRUMViewsPredicate {
     public init () {}
 
     public func rumView(for viewController: UIViewController) -> RUMView? {
-        let canonicalClassName = viewController.canonicalClassName
-        let isCustomClass = canonicalClassName.contains(".") // custom class contains module prefix
-
-        if isCustomClass {
-            var view = RUMView(name: canonicalClassName)
-            view.path = canonicalClassName
-            return view
-        } else {
+        guard !isUIKit(class: type(of: viewController)) else {
+            // do not track bare UIKit's view controllers (UINavigationController, UITabBarController, ...)
```

Comment:
the real goal is excluding container view controllers, exclusion of `UIKit` types is just the workaround that we could come up with.
this comment here may be misleading if we come back to this issue later on, can we correct it?

here is the same issue in PR's description as well:
> ...This was to ignore all the "system" view controllers...

Reply:
No no, it's about plain (not subclassed / customized) VCs from `UIKit`.

Before this PR, this implementation was considering `MyCustomNavigationViewController` as the RUM View. I think we shouldn't change that default if no one has reported it being wrong. We're doing a bug fix here, I wouldn't change the previous behaviour.

Reply:
no i'm not talking about the behavior, my concern is that the comments don't tell the whole story. 
we first wanted to exclude containers and then noticed invisible view controllers in the hiearchy. if one asks "why do we exclude UIKit types?", the answer is not obvious without this story.

IMO this comment `do not track UIKit controllers` brings questions (_why?_) rather than answering

Reply:
I see 👍. Improved it with discussing the reasoning and optimisation.